### PR TITLE
Made clear_progress_widget, moved campaign clear to dialog

### PR DESCRIPTION
**Pull request by TheBeville**
#33

Screenshot 2024-01-16 at 00 09 40

Screenshot 2024-01-16 at 00 10 02

### CODE REVIEW

File: lib/widgets/clear_progress_widget.dart [hunk 1]
```diff
@@ -0,0 +1,90 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:hot_cold/store/progress_store.dart';
+
+class ClearProgressWidget extends StatefulWidget {
```
[the remark this answers is not in the document]
Comment (TheBeville) [reply lib/widgets/clear_progress_widget.dart:5]: Agreed, struggled to settle on a name at the time.

File: lib/widgets/clear_progress_widget.dart [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+  const ClearProgressWidget({super.key});
+
+  @override
+  State<ClearProgressWidget> createState() => _ClearProgressWidgetState();
+}
+
+class _ClearProgressWidgetState extends State<ClearProgressWidget> {
+  bool confirmationDialogShowing = false;
```
[the remark this answers is not in the document]
Comment (TheBeville) [reply lib/widgets/clear_progress_widget.dart:13]: Noted and changed.

File: lib/widgets/clear_progress_widget.dart [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+        ),
+        confirmationDialogShowing
+            ? Padding(
+                padding: const EdgeInsets.only(top: 5.0),
+                child: Row(
+                  mainAxisAlignment: MainAxisAlignment.spaceAround,
+                  children: [
+                    OutlinedButton(
+                      onPressed: () {
+                        context.read<ProgressStore>().clear();
```
[the remark this answers is not in the document]
Comment (TheBeville) [reply lib/widgets/clear_progress_widget.dart:42]: Didn't realise it was in the locator, I'd just lifted the function directly from the code when it was in the appbar haha

File: lib/widgets/clear_progress_widget.dart [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+                      ),
+                      child: const Text(
+                        'Confirm',
+                        style: TextStyle(color: Colors.white),
+                      ),
+                    ),
+                    OutlinedButton(
+                      onPressed: () {
+                        setState(() {
+                          confirmationDialogShowing = false;
```
[the remark this answers is not in the document]
Comment (TheBeville) [reply lib/widgets/clear_progress_widget.dart:60]: Changed, probably a result of coding after midnight while sleepy.

File: lib/widgets/clear_progress_widget.dart [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+                        backgroundColor: Colors.green,
+                      ),
+                      child: const Text(
+                        'Confirm',
+                        style: TextStyle(color: Colors.white),
+                      ),
+                    ),
+                    OutlinedButton(
+                      onPressed: () {
+                        setConfirming(false);
```
Comment (alexobviously) [inline lib/widgets/clear_progress_widget.dart:59]: these could be arrow functions

**Review by alexobviously — APPROVED** [review]
lgtm

File: lib/widgets/clear_progress_widget.dart (hunk 5 above)
Comment (TheBeville) [reply lib/widgets/clear_progress_widget.dart:59]: Changed

**Pull request merged**